### ISSUE TITLE
Tests requiring NuGetSdkResolver not working?

### ISSUE THREAD

**Issue by japj** [jeffkl/MSBuildProjectCreator#48]
It seems Tests cannot resolve NuGet based Sdks, e.g. the Microsoft.Build.NoTargets (NuGet) Sdk.

This can be reproduced by adding the following test to `BuildTests.cs`:
```c#
        [Fact]
        public void CanUseNuGetSdkResolver()
        {
            ProjectCreator
               .Create(Path.Combine(TestRootPath, "project1.proj"))
               .ImportSdk("Sdk.props", "Microsoft.Build.NoTargets", "1.0.53")
               .ImportSdk("Sdk.targets", "Microsoft.Build.NoTargets", "1.0.53")
               .TryRestore(out bool result, out BuildOutput buildOutput);

            result.ShouldBeTrue();
        }
```

This fails with:
```
Message: 
    Microsoft.Build.Exceptions.InvalidProjectFileException : The SDK 'Microsoft.Build.NoTargets' specified could not be found.  C:\Users\japj\AppData\Local\Temp\02qqg4do.avw\project1.proj
  Stack Trace: 
    ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
    ProjectErrorUtilities.ThrowInvalidProject[T1](IElementLocation elementLocation, String resourceName, T1 arg0)
    Evaluator`4.ExpandAndLoadImportsFromUnescapedImportExpressionConditioned(String directoryOfImportingFile, ProjectImportElement importElement, List`1& projects, SdkResult& sdkResult, Boolean throwOnFileNotExistsError)
    Evaluator`4.ExpandAndLoadImports(String directoryOfImportingFile, ProjectImportElement importElement, SdkResult& sdkResult)
    Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
    Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
    Evaluator`4.Evaluate(ILoggingService loggingService, BuildEventContext buildEventContext)
    Evaluator`4.Evaluate(IEvaluatorData`4 data, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
    ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
    ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
    ProjectImpl.ReevaluateIfNecessary(EvaluationContext evaluationContext)
    ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
    Project.ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
    Project.ctor(ProjectRootElement xml, IDictionary`2 globalProperties, String toolsVersion, ProjectCollection projectCollection, ProjectLoadSettings loadSettings)
    ProjectCreator.TryGetProject(Project& project, IDictionary`2 globalProperties, String toolsVersion, ProjectCollection projectCollection, ProjectLoadSettings projectLoadSettings) line 69
    ProjectCreator.get_Project() line 39
    ProjectCreator.TryRestore(Boolean& result, BuildOutput& buildOutput, IDictionary`2& targetOutputs) line 332
    ProjectCreator.TryRestore(Boolean& result, BuildOutput& buildOutput) line 277
    BuildTests.CanUseNuGetSdkResolver() line 141
```

This is with VS2019 16.4.0

I don't know if the NuGetSdkResolver needs to be explicitly enabled, but since dotnet sdk projects seem to work fine, I don't understand why it is not working correctly out of the box.

**Comment by jeffkl** [comment]
This turned out to be a bug in MSBuild which I've filed here: https://github.com/microsoft/msbuild/issues/5320

The only workaround at the moment is you have to save the project and call `RootElement.Reload()`

```cs
ProjectCreator projectCreator = ProjectCreator
    .Create(GetTempFileName(".csproj"), projectCollection: projectCollection)
    .ImportSdk("Sdk.props", "Microsoft.Build.NoTargets", "1.0.53")
    .ImportSdk("Sdk.targets", "Microsoft.Build.NoTargets", "1.0.53")
    .Save(GetTempFileName(".csproj"));

projectCreator.RootElement.Reload();

Project project = projectCreator.Project;
```

**Comment by japj** [comment]
I can confirm that the workaround is ok for me.
Do you want to close this issue or keep it open until the MSBuild one is solved?

**Comment by jeffkl** [comment]
I'll leave this open for now as I might put a workaround in this library.

**Comment by japj** [comment]
If I see correctly then the bug is fixed in the MSBuild 16.7 branch so that will be the first release without needing the workaround

**Comment by jeffkl** [comment]
Closing now that the fix is in MSBuild.
